### DISCUSSION TITLE
Get Past Week's Rosters?

### DISCUSSION THREAD

**Discussion by JimNayzium** [nmelhado/league-page#287] [Help!]
It seems as though the sleeper API lets us grab roster info but it's current to what is live right now.
How can I say for example see my roster information (players/stat totals etc..) for say week 4 in this season?

When I tried to hack into it and do it, it seems as though sleeper API may not even offer this information?
We'd need to save it in our own database week by week it feels like.
Am I way off, here? 

Is there a way to pull a past week roster info like on the sleeper app with "OFF ROSTER" marked type thing? And to be clear I mean for the BENCH AND STARTERS and IR slots as well.

**Comment by nmelhado** [comment]
Hey! Sleeper doesn't _officially_ support this, but a hacky way to get this done would be to use the matchup API instead. What are you looking to do with the past roster info? Adding a new feature?

**Reply by JimNayzium** [reply]
Well I am working on some different things in general that use the Sleeper API and found your awesome stuff!! Your stuff was a wonderful intro for me into the sveltekit world as well!! 

But yeah, I am trying to just see the rosters accurate to the given weeks when they occurred like the sleeper app does with the tag of "off-roster" if they were ON THE roster that week but no longer on the roster. 

I looked at the API for the matchup too and see where I can go that route, but again, I think the easiest way home for what I am kinda thinking would be to just save the rosters each week to my own database and tie them to the sleeper ID and week/season info that way. Seems easy enough.

THANKS!
